core/adapters/http_test.go

```diff
@@ -27,8 +27,8 @@ func TestHttpAdapters_NotAUrlError(t *testing.T) {
 		name    string
 		adapter adapters.BaseAdapter
 	}{
-		{"HTTPGet", &adapters.HTTPGet{URL: cltest.WebURL(t, "NotAURL")}},
-		{"HTTPPost", &adapters.HTTPPost{URL: cltest.WebURL(t, "NotAURL")}},
+		{"HTTPGet", &adapters.HTTPGet{URL: *cltest.WebURL(t, "NotAURL")}},
```

Comment:
It looks like you've changed the `cltest.WebURL` helper to return a pointer instead of the an actual instance, and this has trickled into all the other tests that use that helper. I think a better strategy would be to just take the pointer of the instance when you need it, then you won't need to add all these extra asterisks everywhere.